python/pyspark/ml/feature.py

```diff
@@ -1330,6 +1448,21 @@ class StringIndexer(JavaEstimator, HasInputCol, HasOutputCol, HasHandleInvalid):
     >>> sorted(set([(i[0], str(i[1])) for i in itd.select(itd.id, itd.label2).collect()]),
     ...     key=lambda x: x[0])
     [(0, 'a'), (1, 'b'), (2, 'c'), (3, 'a'), (4, 'a'), (5, 'c')]
+    >>> stringIndexerPath = temp_path + "/string-indexer"
+    >>> stringIndexer.save(stringIndexerPath)
+    >>> loadedIndexerModel = StringIndexer.load(stringIndexerPath).fit(stringIndDf)
```

Comment:
It does not need to train here. Like others, check ```StringIndexer``` param equality, then save the model trained at L1323, loaded the model and check model equality.